src/setup-dotnet.ts

```diff
@@ -51,21 +49,47 @@ export async function run() {
   }
 }
 
-function getVersionFromGlobalJson(globalJsonPath: string): string {
-  let version: string = '';
+function getVersionFromGlobalJson(hasVersions: boolean): string | null {
```

Comment:
Personally, I think the logic with `hasVersions` variable is not clear enough and difficult from readability side because on the first sight it looks like we read globalJson file always.
I think it will be easier to read if we move the condition to parent function (`run`)
Something like that (pseudocode, not tested)
```ts
let versions = core.getMultilineInput('dotnet-version');
const globalJsonFileInput = core.getInput('global-json-file');
if (versions.length === 0 || globalJsonFileInput) {
  const globalJsonVersion = getVersionFromGlobalJson(globalJsonFileInput);
  if (globalJsonVersion) {
    versions.push(globalJsonVersion);
  }
}
...
function getVersionFromGlobalJson(globalJsonFileInput: string | null): string | null {
```

This way it will be clearer that we read globalJsonFile only in case if 0 versions are specified or explicit global json file path is defined.

Reply:
I believe it's important that we throw if you specified `global-json-file` but it doesn't exist. This is the least surprising behaviour (I could imagine trying to track down why the wrong version was being installed in my future if it didn't), and it aligns with the behaviour in setup-node.

I was also trying to maintain the existing debug output, but can take that as being unnecessary.

Perhaps just something like this (pseudo)? Changes a bit less, clutters the main path a bit more, but it will state the flow clearer.

```js
let versions = core.getMultilineInput('dotnet-version');

const globalJsonFileInput = core.getInput('global-json-file');
if (globalJsonFileInput) {
  if (!existsSync(globalJsonFileInput)) {
    throw new Error('...');
  }
  versions.push(getVersionFromGlobalJson(globalJsonFileInput));
}

if (versions.length === 0) {
  core.debug('No version found, trying to find version from global.json');
  if (existsSync('global.json')) {
    versions.push(getVersionFromGlobalJson('global.json'));
  }
}

Reply:
yes, looks good to me! Thanks for proposing

Reply:
@omsmith  @maxim-lobanov, do we definitely want to install dotnet version from global.json as additional version?
It's a bit of a contradiction the previous logic: if `dotnet-version` is not supplied, look a version in global.json

Reply:
I think if it's specified we should definitely attempt to use it. In setup-node it's an error to specify both `node-version` and `node-version-file`, but it also only supports a single version in general.

Given dotnet supports multiple version, and we support that here with multiline `dotnet-version` I thought it made sense not to make them mutually exclusive.

If I were to use both, I would probably treat my specified global.json file as my primary and dotnet-version as additional ones I'm installing.

Reply:
I agree that if `global-json-file` is defined, we can't ignore it silently. So we have two options if both `dotnet-version` and `global-json-file` are defined:
- Throw exception and fail task with message that options can't be mixed.
- Use both `dotnet-version` and `global-json-file` values

The second option looks a bit more friendly and I don't see significant problems with it. But I agree that it is pretty rare case to use :)

Reply:
Just to clarify - I mean only case when both `dotnet-version` and `global-json-file` are specified. 
If `global-json-file` is not defined and `dotnet-version` is defined, we shouldn't look at default `global.json`

Reply:
Thanks for the feedback. I've rebased and force-pushed the update since it's reverting most of the changes I'd done.